### PR TITLE
add tracking with a buffer of latest N frames

### PR DESCRIPTION
**Pull request by simogasp**

### CODE REVIEW

**Comment by simogasp** [comment]
need update

File: src/software/Localization/main_cameraLocalizer.cpp [hunk 1]
```diff
@@ -173,6 +173,8 @@ int main(int argc, char** argv)
   std::string visualDebug = "";        ///< whether to save visual debug info
   bool useVoctreeLocalizer = true;        ///< whether to use the voctreeLocalizer or cctagLocalizer
   bool useSIFT_CCTAG = false;             ///< whether to use SIFT_CCTAG
+  
+  bool useFrameBufferMatching = true;     ///< enable the matchign with the last N frame of the sequence
```
Comment (pmoulon) [inline src/software/Localization/main_cameraLocalizer.cpp]: matchign => matching ;-)

File: src/openMVG/localization/BoundedBuffer.hpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  /**
+   * @brief Returns an iterator pointing to the last element of the buffer.
+   * 
+   * @return the iterator pointing to the last element of the buffer.
+   */
+  const_iterator end() const { return _buffer.end(); }
+  
+  /**
+   * @brief It add a new element at the end of the buffer. If the buffer is full
```
Comment (simogasp) [inline src/openMVG/localization/BoundedBuffer.hpp]: adds

File: src/software/Localization/main_cameraLocalizer.cpp [hunk 3]
```diff
@@ -236,6 +238,8 @@ int main(int argc, char** argv)
           "[voctree] Maximum matching error (in pixels) allowed for image matching with "
           "geometric verification. If set to 0 it lets the ACRansac select "
           "an optimal value.")
+      ("useFrameBufferMatching", po::bool_switch(&useFrameBufferMatching),
```
Comment (fabiencastan) [inline src/software/Localization/main_cameraLocalizer.cpp]: Maybe more useful to expose N directly instead of having 2 parameters (in all the code)?

Comment (simogasp) [inline src/software/Localization/main_cameraLocalizer.cpp]: yes. So far the size of `_frameBuffer` is hardcoded to 5 in the class constructor of VoctreeLocalization. That needs to be changed

**Pull request merged**